### PR TITLE
Fix some hyperlinks

### PR DESCRIPTION
**Pull request by stantios**
Description: This PR fixes hyperlinks in the README file. I have also found other hyperlinks that are broken but I am not sure where they could be redirected to. One more thing, I have also changed the CurseForge link to remove the '/placeholder' since it was an invalid link. Let me know if you want me to revert that change back since I would be happy to.

The ones that are broken are:

- ForgeDev
- Pull-Requests

I found that these two links did not have a page to redirect to or I could not find a page to redirect them to on the current wiki, so I left those hyperlinks alone.

### CODE REVIEW

**Comment by stantios** [comment]
For now, I have redirected the broken hyperlinks to the Forge Documentation until we have pages in the NeoForge Documentation for those hyperlinks to redirect to.

Hope you don't mind.

**Comment by sciwhiz12** [comment]
Please don't link to the Minecraft Forge documentation site in the NeoForge repository. I'd rather we remove/comment out the link temporarily rather than redirect it to Forge, which wouldn't fit our new contributing flow due to the new NeoGradle setup and current (and upcoming) changes to the contributing flow.

**Comment by stantios** [comment]
Oh ok. That's my bad, sorry. I will revert the changes now, and will most likely remove the links so they can be added back later when we have the pages set up.

**Comment by stantios** [comment]
I have now removed the traces of the Minecraft Forge Documentation and removed parts of the wiki that related to the broken hyperlinks. We can add them back with fixed links when we have setup the related pages for them in the NeoForge docs.

Edit: Forgot to mention another change, but I also fixed a grammar error with the word open collective.

File: docs/README.md [hunk 1]
```diff
@@ -31,17 +31,8 @@ For support and questions, visit [the NeoForged Discord server][Discord].
 
 [See the "Getting Started" section in the NeoForged Documentation][Getting-Started].
 
-# Contribute to Forge
```
Comment (sciwhiz12) [inline docs/README.md:34]: This header should be kept, to separate the below sections on pull requests and the CLA from the 'Creating Mods' header. The first paragraph should also be kept, and references to Forge should be renamed to NeoForge.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 1 above)
Comment (stantios) [reply docs/README.md:34]: This should be fixed in this commit (https://github.com/neoforged/NeoForge/pull/202/commits/46a1f85720a61bac365142e47816399e8987d1ea).

**Comment by stantios** [comment]
For some reason, it is saying that I haven't signed the CLA but I clearly have as shown here:

image

File: docs/README.md [hunk 2]
```diff
@@ -31,38 +31,29 @@ For support and questions, visit [the NeoForged Discord server][Discord].
...
+### Contribute to NeoForge
```
Comment (sciwhiz12) [inline docs/README.md]: The level of this header should be equal to the level of the `Creating Mods` header. Either this header should be H1 (`#`), or this and the other headers above should be H2 (`##`).

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Please rewrite your commits to have the same consistent email, for the CLA checker's sake.

File: docs/README.md (hunk 2 above)
Comment (stantios) [reply docs/README.md]: I have changed this header and the others to H2 since I feel like it would be a better choice.

**Comment by stantios** [comment]
> Please rewrite your commits to have the same consistent email, for the CLA checker's sake.

I recently changed my email address so I had to setup commit signing again with the right email, so I think it has something to do with that.

**Comment by stantios** [comment]
I am going to close this PR and start from the beginning, if you don't mind.

**Comment by sciwhiz12** [comment]
> > Please rewrite your commits to have the same consistent email, for the CLA checker's sake.
> 
> I recently changed my email address so I had to setup commit signing again with the right email, so I think it has something to do with that.

Yeah, that's probably it. You'll need to rewrite all your commits in this branch to have your new email address on it. Once you do that on your local branch, force-push your local branch to the PR branch.

Or, alternatively (as you've just done), you can redo the PR from the start. If you do, please link to this PR from that new PR's description.

**Comment by stantios** [comment]
Yep, I will.

**Pull request closed without merging**